packages/apps/web-viewer-test/src/providers/EcsqlWidget.tsx

```diff
@@ -0,0 +1,147 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useActiveIModelConnection } from "@itwin/appui-react";
+import { QueryRowFormat } from "@itwin/core-common";
+import type { IModelConnection } from "@itwin/core-frontend";
+import {
+  Button,
+  ExpandableBlock,
+  LabeledTextarea,
+  Table,
+} from "@itwin/itwinui-react";
+import React, { useState } from "react";
+
+export default function EcsqlWidget() {
+  const [input, setInput] = useState("");
+  const [data, setData] = useState<Record<string, string | React.ReactNode>[]>(
+    []
+  );
+  const [headers, setHeaders] = useState<string[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+  const iModelConnection = useActiveIModelConnection() as IModelConnection;
+
+  const getData = async () => {
+    try {
+      setData([]);
+      setHeaders([]);
+      setIsLoading(true);
+      const results = iModelConnection.query(input, undefined, {
```

Comment:
do a null check for iModelConnection right before the try block to make sure its defined,